src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2541,6 +2541,11 @@ object Desugar {
                 conv: in.EffectfulConversion = in.EffectfulConversion(target, resT, dArg)(src)
                 _ <- write(conv)
               } yield target
+            case (t: InterfaceT, _) =>
```

Comment:
Is this already covered by another unit test? It seems that the newly added unit test does not trigger this code at all

Reply:
`client1` covers this code. Nonetheless, I added another test (`client3`) that more clearly relies on this feature.